types/node/index.d.ts

```diff
@@ -1664,7 +1664,7 @@ declare module "https" {
         servername?: string;
     }
 
-    export interface AgentOptions extends http.AgentOptions {
+    export interface AgentOptions extends http.AgentOptions, tls.ConnectionOptions {
         pfx?: any;
```

Comment:
I think `pfx`, `key`, `passphrase`, `cert`, `ca`, `ciphers`, `secureProtocol`, `rejectUnauthorized` and `serverName` should be removed here as they are already part of `tls.ConnectionOptions`.
The difference in typings of `pfx`, `key`, `cert` and `ca` here compared to "tls" is maybe because typing in "tls is incomplete. According to docs it should be like this I think:
`pfx?: string | Buffer | Array<string | Buffer> | Array<{ buf: string | Buffer, passphrase?: string }>`
`key?: string | Buffer | Array<string | Buffer> | Array<{ pem: string | Buffer, passphrase?: string }>`
`cert?: string | Buffer | Array<string | Buffer>`
`ca?: string | Buffer | Array<string | Buffer>`

Reply:
ok. I've removed these options from this interface and updated the type types in SecureContextOptions to match the docs.